recipe-client-addon/lib/RecipeRunner.jsm

```diff
@@ -81,6 +81,20 @@ this.RecipeRunner = {
       await ClientEnvironment.getClientClassification();
     }
 
+    // Run pre-execution hooks. If a hook fails, we don't run recipes with that
+    // action to avoid inconsistencies.
+    const actions = await NormandyApi.fetchActions();
+    for (const action of Object.values(actions)) {
+      try {
+        await this.runActionCallback(action, "preExecution");
```

Comment:
We should make very clear in the docs that preExecution hooks, if any, will run even on builds where filters don't match, as that might surprise people who didn't expect to have to deal with backward compat in the future, or with windows environments when they wrote something for mac users, or...